tests/test_resolver.py

```diff
@@ -45,6 +47,31 @@
 ]
 
 
+@pytest.fixture()
+def check_no_lingering_resolvers() -> Generator[None, None, None]:
+    """Verify no resolvers remain after the test.
+
+    This fixture should be used in any test that creates instances of
+    AsyncResolver or directly uses _DNSResolverManager.
+    """
+    try:
+        yield
+    finally:
+        # We need to check in a finally block to ensure the check runs
+        # even if the test fails
```

Comment:
If the test fails, do we need a check to make it fail..?

Reply:
The test will still fail but if we don't run this fixture's check in the finally block it won't actually run because it already failed

Reply:
Yeah, I'm just wondering if it needs to run? The comment makes it sound like you'd do something like this in a test:
```
try:
    assert something
finally:
    assert second_thing
```

I'd expect if the test has failed, there's no point running further parts of the test.

Reply:
Yeah I guess it doesn't matter since we aren't cleaning up, only asserting that we are cleaned up

Reply:
dropped the finally in 7f54a9db3